kohvis/src/util.rs

```diff
@@ -0,0 +1,19 @@
+use crate::Kohonen;
+use gdnative::{Control, Instance, Node, NodePath};
+
+pub fn get_kohonen_node(owner: Control, path: &str) -> Option<Node> {
+    unsafe { owner.get_node(NodePath::from_str(path)) }
+}
+
+pub fn with_kohonen<F, U>(
+    owner: gdnative::Control,
+    node: Node,
+    fun: F,
+) -> Result<U, gdnative::user_data::LocalCellError>
+where
+    F: FnOnce(gdnative::Control, &Kohonen) -> U,
+{
+    Instance::<Kohonen>::try_from_base(node)
+        .unwrap()
+        .map(|koh, _| fun(owner, koh))
+}
```

Comment:
@adamreichold Still not enough understanding of Rust generics to make this function generic, i.e. usable for everything that inherits NativeClass, not just Kohonen.

Reply:
Maybe I am not seeing something, but from a first glance, I'd have said that
```
pub fn with_something<T, F, U>(
    owner: gdnative::Control,
    node: Node,
    fun: F,
) -> Result<U, gdnative::user_data::LocalCellError>
where
    T: NativeClass
    F: FnOnce(gdnative::Control, &T) -> U,
{
    Instance::<T>::try_from_base(node)
        .unwrap()
        .map(|koh, _| fun(owner, koh))
}
```
should do the trick based on the type signature of [`Instance`](https://docs.rs/gdnative/0.8.1/gdnative/struct.Instance.html#implementations)?

But then again, such a method is so generic that it almost adds nothing over using `Instance` direclty and the type interference is probably a bit cumbersome, e.g. calling via `with_something::<Kohonen, _, _>(owner, node, |owner, koh| ...);` or `with_something(owner, node, |owner, koh: &Kohonen| ...);`.